docs/build/js-client/10-marine-js.md

```diff
@@ -1,63 +1,138 @@
+Accordingly, each user has its own JS client up and running.
+JS client hosts logic (in a dedicated service or services) for interacting with incoming particles.
+Each particle carries a message of new typed symbol or added exclamation mark and more,
+and the client contains logic to process it.
 
-This function can only be used in nodejs. Trying to call it inside a browser will throw an error.
+### Example 2
 
-**loadWasmFromNpmPackage**
+AI image library with an ability to execute incoming processing requests and distribute the work through Fluence network.
+The lib should do some simple processing locally, e.g., directly in a browser and also parallel the work if necessary.
 
-Locates a WASM file in the specified npm package and loads it. The function accepts two arguments:
+### Summary
 
-- Name of an npm package
-- Path to a WASM file relative to the npm package
+These use cases have something in common—they need to host a logic locally.
+Moreover, sometimes writing a logic in pure JavaScript is not feasible.
+For example, AI processing logic would be a lot effective if written in low-level language like Rust
 
-This function can only be used in nodejs. Trying to call it inside a browser will result in an error.
+## Use cases
 
-**loadWasmFromServer**
+Usually you don't need to interact with Marine JS directly except the following:
+- Cross-platform service which works both on Nox peer and JS client peer
+- Heavy computations, CPU intensive tasks
+- Would prefer to write service in Rust or other language with compilation to WASM
 
-Loads a WASM file from the service hosting the application. It accepts the file path on server and returns a buffer compatible with the `FluencePeer` API.
+In any matched case, writing your service as WASM more suitable than doing so in JavaScript.
 
-:::info
-The function will try to load a file into SharedArrayBuffer if the site is cross-origin isolated. Otherwise, the return value falls back to Buffer, which is less performant but is still compatible with the `FluencePeer` API.
+## What is Marine JS
 
-We strongly recommend to set-up cross-origin headers. For more details, see the [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer#security_requirements).
-:::
+When JS client starts up, it starts with initialization Marine JS runtime.
+The runtime hosts wasm services and even allows you to register your own wasm services.
+For example, AquaVM service for processing particles and Air code resides in Marine JS.
+That's why Marine JS is a foundation of JS client.
 
-This function can only be used in a browser. Trying to call it inside nodejs will result in an error.
+## Extending JS client with Marine services
 
-### Registering services in FluencePeer
+> Currently only [pure](https://fluence.dev/docs/build/glossary#pure-module) single-module services are supported.
```

Comment:
I remember raising this question before already, but I just to confirm. I think Valery made multi-module services working. Why is this still saying only single-module are supported?

Reply:
There is a task for supporting multi-module services in JS client. Currently it's not supported and doing that is a heavy task.